### PR TITLE
external-links: Add icons to external links

### PR DESCRIPTION
**Pull request by Xenosb**
Improve style of external links

References #112 

image

### CODE REVIEW

**Comment by Macxim** [comment]
We made a PR to actually remove icon fonts 😅

File: app/views/layouts/_links.html.erb [hunk 1]
```diff
@@ -10,21 +10,50 @@
...
-          Pomoć obiteljima iz Petrinje, Siska i okolice - kontakti, smještaj i prijevoz - Google doc (preview)
-        </a></li>
-        <li><a href="https://docs.google.com/document/d/1yUxHtKCkkXEQ8tXTD2bODbisxwF5Q9zvPJoU27vQGUM/edit?fbclid=IwAR0g1F__7avcVpOlf__NQgBq626LDmDNyz42k545TT5MCjTLMvf4De9tvMs" target="_blank">
-          Pomoć za Petrinju - Google doc
-        </a></li>
-        <li><a href="https://www.facebook.com/groups/155412969344431/permalink/155812429304485/" target="_blank">
-          POMOC GLINI, PETRINJI, SISKU i okolnim mjestima - Facebook grupa
-        </a></li>
-        <li><a href="https://potres2020.openit.hr/views/map" target="_blank">
+      <div class="is-full my-1">
```
Comment (Macxim) [inline app/views/layouts/_links.html.erb:25]: Why did you remove the `<ul>` ?

**Comment by Xenosb** [comment]
😺 oupsey about icon fonts! What was the reason to remove them? Lower page load time?

File: app/views/layouts/_links.html.erb (hunk 1 above)
Comment (Xenosb) [reply app/views/layouts/_links.html.erb:25]: List was adding bullets and it somehow seemed easier to remove it and add divs. But in can revert it to `<ul>` and add scss changes so that bullets are not added.

Comment (Macxim) [reply app/views/layouts/_links.html.erb:25]: Yes, please do.

**Comment by Macxim** [comment]
About icon fonts:

- https://css-tricks.com/icon-fonts-vs-svg/
- https://github.blog/2016-02-22-delivering-octicons-with-svg/
- https://www.ianfeather.co.uk/ten-reasons-we-switched-from-an-icon-font-to-svg/
- https://cloudfour.com/thinks/seriously-dont-use-icon-fonts/

File: app/views/layouts/_links.html.erb (hunk 1 above)
Comment (Xenosb) [reply app/views/layouts/_links.html.erb]: Solved. Please check!

**Comment by Xenosb** [comment]
I could probably add needed icons manually or fetch them from somewhere but I'd like something that's easier to implement. Any suggestions?

**Comment by vlado** [comment]
@Xenosb See https://github.com/vlado/earthquake-croatia/pull/82 for icons

**Comment by Xenosb** [comment]
Ready for re-review

File: app/views/layouts/_links.html.erb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <div id="external-links" class="card mb-6" data-controller="card">
   <a href="javascript:void(0)" class="card-header" data-action="card#toggle" aria-label="hide links">
     <span class="card-header-title">
-      Vanjske poveznice s popisima ponude smještaja i prijevoza
+      Kako mogu pomoći?
```
Comment (vlado) [inline app/views/layouts/_links.html.erb:4]: Not sure if "Kako mogu pomoći" is the right content but do not have anything better.

**Review by vlado — APPROVED** [review]
Looks good to me 👍 

@Macxim please check and if all is ok we can merge.

**Comment by vlado** [comment]
@Macxim here is the screenshot

Screenshot 2021-01-04 at 11 53 37

**Comment by Macxim** [comment]
I made a few comments. Please check. Let me know if you need help.

**Comment by Xenosb** [comment]
> I made a few comments. Please check. Let me know if you need help.

Can you please point out to unresolved ones?

**Comment by vfonic** [comment]
I'm also not a fan of phrasing "Kako mogu pomoći?" ("How can I help?"). It opens up an HTML container with various links inside. Is that the only links/places where I can go help? What about this very website? It's not there. That's really weird.

I liked the original phrasing. Or something along the lines of "Ostale poveznice gdje se može pomoći"...

Are we building this app to be the last place people come for offering/asking help or to be the first place?

**Comment by Xenosb** [comment]
I'm preparing this as a follow-up MR

**Review by Macxim — CHANGES_REQUESTED** [review]

**Comment by vfonic** [comment]
@Xenosb follow-up for what? Sorry I'm not 100% up to date.

File: app/assets/stylesheets/external-links.scss [hunk 3]
```diff
@@ -0,0 +1,20 @@
+
+#external-links {
```
[the remark this answers is not in the document]
Comment (Xenosb) [reply app/assets/stylesheets/external-links.scss:2]: Good tip, will implement!

File: app/assets/stylesheets/external-links.scss [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+  .media {
+    .icon {
+      fill: #4a4a4a;
+    }
+  }
+
+  li {
+    margin-bottom: 0.5rem;
```
[the remark this answers is not in the document]
Comment (Xenosb) [reply app/assets/stylesheets/external-links.scss:11]: I was not sure if it was better to add `mb-2` to each element of the list or to add a css rule. For me css sounded as a more logical solution. You only need to change one line to affect all of the list elements while in case of helpers you have a possibility of forgetting and introducing bugs.

File: app/assets/stylesheets/external-links.scss [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+      .icon {
+        vertical-align: bottom;
```
[the remark this answers is not in the document]
Comment (Xenosb) [reply app/assets/stylesheets/external-links.scss:14]: Can you please provide a code suggestion?

**Comment by Xenosb** [comment]
@vfonic 

I was on the road and replied using email but the attachments didn't show up here for some reason.
I'm planning to add more content to `Kako pomoci` card. For example, adding links on how to donate will close #43 , #45 and #51

blah

**Comment by vfonic** [comment]
Great! I like that! Thanks for clarifying!

File: app/assets/stylesheets/external-links.scss (hunk 5 above)
[the remark this answers is not in the document]
Comment (vfonic) [reply app/assets/stylesheets/external-links.scss:14]: Sorry to hijack this @Macxim. Feel free to add/fix if I miss anything. I'm writing this from the top of my head:

```css
.parent-class {
  display: flex;
  // flex-direction: row; // this is the default that's why it's commented out, just to show the default value
  align-items: flex-end;
}
```

Or, if you only want to align the icon to bottom:

```css
.icon { // please make sure this remains nested (not too many levels nested), as it's very generic class name
  align-self: flex-end;
}
```

**Comment by Macxim** [comment]
This is getting quite big for a dropdown. Next step will be moving all this to a separate menu.

File: app/assets/stylesheets/external-links.scss (hunk 4 above)
[the remark this answers is not in the document]
Comment (Xenosb) [reply app/assets/stylesheets/external-links.scss:11]: @Macxim what's your take on this - single change of css or adding helpers to all menus?

File: app/assets/stylesheets/external-links.scss (hunk 3 above)
[the remark this answers is not in the document]
Comment (Xenosb) [reply app/assets/stylesheets/external-links.scss]: Implemented

File: app/assets/stylesheets/external-links.scss (hunk 4 above)
[the remark this answers is not in the document]
Comment (Xenosb) [reply app/assets/stylesheets/external-links.scss]: I've changed from css changes to adding helpers

File: app/assets/stylesheets/external-links.scss (hunk 5 above)
[the remark this answers is not in the document]
Comment (Xenosb) [reply app/assets/stylesheets/external-links.scss]: Implemented, please check

File: app/assets/images/hand-holding-heart.svg [hunk 6]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512"><!-- Font Awesome Free 5.15.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) --><path d="M275.3 250.5c7 7.4 18.4 7.4 25.5 0l108.9-114.2c31.6-33.2 29.8-88.2-5.6-118.8-30.8-26.7-76.7-21.9-104.9 7.7L288 36.9l-11.1-11.6C248.7-4.4 202.8-9.2 172 17.5c-35.3 30.6-37.2 85.6-5.6 118.8l108.9 114.2zm290 77.6c-11.8-10.7-30.2-10-42.6 0L430.3 402c-11.3 9.1-25.4 14-40 14H272c-8.8 0-16-7.2-16-16s7.2-16 16-16h78.3c15.9 0 30.7-10.9 33.3-26.6 3.3-20-12.1-37.4-31.6-37.4H192c-27 0-53.1 9.3-74.1 26.3L71.4 384H16c-8.8 0-16 7.2-16 16v96c0 8.8 7.2 16 16 16h356.8c14.5 0 28.6-4.9 40-14L564 377c15.2-12.1 16.4-35.3 1.3-48.9z"/></svg>
```
Comment (vfonic) [inline app/assets/images/hand-holding-heart.svg:1]: Is this icon still needed/used?

File: app/assets/stylesheets/application.scss [hunk 7]
```diff
@@ -14,6 +14,7 @@
  *= require_self
  */
 
+@import "external-links";
```
Comment (vfonic) [inline app/assets/stylesheets/application.scss:17]: Looks like this is not necessary as `require_tree .` already imports it. (`require_tree .` => require all the files and directories recursively from this file's directory)

We might decide to remove `require_tree .` if we want to use SCSS variables across multiple files though.

File: app/assets/stylesheets/external-links.scss [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+  .media {
+    .icon {
+      fill: #4a4a4a;
+    }
+  }
+
+  li {
+    display: flex;
+    align-items: flex-end;
```
Comment (vfonic) [inline app/assets/stylesheets/external-links.scss:12]: Sorry to make you change this one again!

I just realized that on mobile it looks kinda broken:

* `align-items: flex-end`:
  Screen Shot 2021-01-05 at 18 41 55
* `align-items: center`:
Screen Shot 2021-01-05 at 18 42 02
* `align-items: flex-start`:
Screen Shot 2021-01-05 at 18 42 11


I'd say `align-items: flex-start` looks the best.

**Review by vfonic — CHANGES_REQUESTED** [review]
Thanks! This looks really good overall. I just noticed one issue on mobile.

**Comment by Macxim** [comment]
Please ping me when you're done to review again.

File: app/assets/images/hand-holding-heart.svg (hunk 6 above)
Comment (Xenosb) [reply app/assets/images/hand-holding-heart.svg:1]: Not in this MR, removing it.

File: app/assets/stylesheets/application.scss (hunk 7 above)
Comment (Xenosb) [reply app/assets/stylesheets/application.scss:17]: Works without includes of `hero` and `external-links` so I've removed both of 'em

File: app/assets/stylesheets/external-links.scss (hunk 8 above)
Comment (Xenosb) [reply app/assets/stylesheets/external-links.scss:12]: Good catch, thanks! Agree with you that `flex-start` looks the best.

File: app/assets/stylesheets/external-links.scss (hunk 4 above)
[the remark this answers is not in the document]
Comment (vfonic) [reply app/assets/stylesheets/external-links.scss]: I think, with the rise of utility-based CSS frameworks, the idea is to write the least amount of pure CSS as possible. Basically the restrictions are there for reason: to prevent users from doing: `font-size: 12px`, next to a smaller text of: `font-size: 9px`, next to a sightly bigger text of `font-size: 10px`, but then that one button should be something in between, so let's add `font-size: 11px` only here.

It's also easier to open `.html.erb` file and see this:

```html
<a class="button is-primary is-outlined m-2 px-4 py-2">
  Subscribe
</a>
```

Than to see this:
```html
<a class="subscribe-button button is-primary is-outlined m-2 py-2">
  Subscribe
</a>
```

```css
.subscribe-button {
  padding-left: 10px;
  padding-right: 10px;
}
```

That's at least how I see it. Tailwind CSS is a leader in this "school of thought". :)

**Review by vfonic — APPROVED** [review]
Looks good to me! :shipit:

**Comment by Xenosb** [comment]
@Macxim I've pushed suggestions from @vfonic. Please take a look 🔍

File: app/views/layouts/_links.html.erb [hunk 9]
```diff
@@ -10,20 +10,51 @@
...
-        </a></li>
+    <div class="help-category">
+      <div class="media">
+        <div class="media-left">
+          <span class="icon is-large">
+            <%= inline_svg_tag('globe.svg', aria: true) %>
+          </span>
+        </div>
+        <div class="media-content">
+          <p class="title is-4">Vanjske poveznice</p>
```
Comment (Macxim) [inline app/views/layouts/_links.html.erb:21]: A title should be a `<h*>`, not a `<p>` tag.

File: app/views/layouts/_links.html.erb [hunk 10]
```diff
@@ -10,20 +10,51 @@
...
+          <p class="subtitle is-6">Provjerene poveznice na korisne dokumente i web stranice</p>
+        </div>
+      </div>
+      <ul class="content pl-2">
+        <li class="my-1 mr-2">
```
Comment (Macxim) [inline app/views/layouts/_links.html.erb:26]: Why do we need a right margin on the `<li>`?

File: app/views/layouts/_links.html.erb [hunk 11]
```diff
@@ -10,20 +10,51 @@
...
+          <span class="icon mr-1">
+            <%= inline_svg_tag('file-word.svg', aria: true) %>
+          </span>
+          <a href="https://docs.google.com/document/d/15kSw0O1jtZYTeLh3jZg2XeGFbNbavIBsA0u_PvK8YRs"  target="_blank">
+            Pomoć obiteljima iz Petrinje, Siska i okolice - kontakti, smještaj i prijevoz
+          </a>
+        </li>
+        <li class="my-1 mr-2">
+          <span class="icon mr-1">
```
Comment (Macxim) [inline app/views/layouts/_links.html.erb:35]: For consistency, let's use a `mr-2` spacing after an icon.

File: app/views/layouts/_links.html.erb [hunk 12]
```diff
@@ -10,20 +10,51 @@
...
+            <%= inline_svg_tag('file-word.svg', aria: true) %>
+          </span>
+          <a href="https://docs.google.com/document/d/1yUxHtKCkkXEQ8tXTD2bODbisxwF5Q9zvPJoU27vQGUM" target="_blank">
+            Pomoć za Petrinju
+          </a>
+        </li>
+        <li class="my-1 mr-2">
+          <span class="icon mr-1">
```
Comment (Macxim) [inline app/views/layouts/_links.html.erb:43]: Same as above.

File: app/views/layouts/_links.html.erb [hunk 13]
```diff
@@ -10,20 +10,51 @@
...
+            <%= inline_svg_tag('facebook-square.svg', aria: true) %>
+          </span>
+          <a href="https://www.facebook.com/groups/155412969344431/permalink/155812429304485/" target="_blank">
+            Pomoć Glini, Petrinji, Sisku i okolnim mjestima
+          </a>
+        </li>
+        <li class="my-1 mr-2">
+          <span class="icon mr-1">
```
Comment (Macxim) [inline app/views/layouts/_links.html.erb:51]: Same as above.

File: app/assets/stylesheets/external-links.scss [hunk 14]
```diff
@@ -0,0 +1,18 @@
+
```
Comment (Macxim) [inline app/assets/stylesheets/external-links.scss:1]: I don't think we even need this file at all. You should use flexbox helpers whenever possible.
See: https://bulma.io/documentation/helpers/flexbox-helpers/

File: app/views/layouts/_links.html.erb [hunk 15]
```diff
@@ -10,20 +10,51 @@
...
-          POMOC GLINI, PETRINJI, SISKU i okolnim mjestima - Facebook grupa
-        </a></li>
-        <li><a href="https://potres2020.openit.hr/views/map" target="_blank">
-          Potres 2020 - mapa pomoći
-        </a></li>
+    <div class="help-category">
+      <div class="media">
+        <div class="media-left">
+          <span class="icon is-large">
+            <%= inline_svg_tag('globe.svg', aria: true) %>
```
Comment (Macxim) [inline app/views/layouts/_links.html.erb:17]: Not sure if we need yet another icon for this.

**Review by Macxim — CHANGES_REQUESTED** [review]
I made a few comments.

Regarding external links, I don't think we should add `target='_blank'`, see: https://css-tricks.com/use-target_blank/

File: app/views/layouts/_links.html.erb (hunk 15 above)
Comment (Xenosb) [reply app/views/layouts/_links.html.erb:17]: Which already available icon would you use?

File: app/assets/stylesheets/external-links.scss (hunk 14 above)
Comment (Xenosb) [reply app/assets/stylesheets/external-links.scss:1]: I've removed everything except `fill: #4a4a4a;` From what I saw in Bulma docs, color and icon helpers are modifying `color` and we need to modify `fill`.

As a solution to eliminate the file, I can add a new class eg `fill-gray` to `application.scss` and assign icons to it. Would that be ok?

Comment (vfonic) [reply app/assets/stylesheets/external-links.scss]: It might work even if you just change the color. (Maybe bulma sets `fill: currentColor`, which picks up the color from the `color` property.)

Comment (Xenosb) [reply app/assets/stylesheets/external-links.scss]: I've tried that and it doesn't seem to work. But I noticed that it works for map icon in ad details. Took me some time but I've figured out that this is because I'm using fontawesome solid icons which have no stroke color but fill the inside of the shape.

I'll check out how regular icons look and post a screenshot here shortly

Comment (vfonic) [reply app/assets/stylesheets/external-links.scss]: If you feel adventurous, you could try adding `fill="currentColor"` in the svg files in the `<path>` tag. :)

Comment (Xenosb) [reply app/assets/stylesheets/external-links.scss]: That worked like a charm! Good tip 🦑

Comment (Macxim) [reply app/assets/stylesheets/external-links.scss]: Out of curiosity, have you tried adding a text colour class to

File: app/views/layouts/_links.html.erb (hunk 15 above)
Comment (Macxim) [reply app/views/layouts/_links.html.erb:17]: No icon at all?

File: app/assets/stylesheets/external-links.scss (hunk 14 above)
Comment (Xenosb) [reply app/assets/stylesheets/external-links.scss]: Yup. Didn't work.

File: app/views/layouts/_links.html.erb (hunk 15 above)
Comment (Xenosb) [reply app/views/layouts/_links.html.erb:17]: To me it looks a bit bland without the icon

image

**Comment by vlado** [comment]
Coming late to the party :)

Can we add link to potres-help.com?

**Comment by Xenosb** [comment]
> Coming late to the party :)
> 
> Can we add link to potres-help.com?

Sure! I'll add it right now

**Comment by vlado** [comment]
> > Coming late to the party :)
> > Can we add link to potres-help.com?
> 
> @vlado Sure! I'll add it right now

Thanks 👍
Can we merge and deploy after that?

Great work everyone. Didn't mean it will take 4 days :) but it really is awesome now.

**Comment by Xenosb** [comment]
@vlado pushed the change with link to *potres-help.com*. I believe you can merge. Only open discussion is about removal of the icon which is styling and can be addressed later.

image
